llvm/lib/CodeGen/SelectionDAG/TargetLowering.cpp

```diff
@@ -3736,6 +3736,14 @@ bool TargetLowering::SimplifyDemandedVectorElts(
       KnownUndef.clearAllBits();
     }
     break;
+  case ISD::SINT_TO_FP:
+  case ISD::UINT_TO_FP:
+  case ISD::FP_TO_SINT:
+  case ISD::FP_TO_UINT:
+    if (SimplifyDemandedVectorElts(Op.getOperand(0), DemandedElts, KnownUndef,
+                                   KnownZero, TLO, Depth + 1))
+      return true;
+    break;
```

Comment:
Is the fallthrough to the undef handling tested (not sure this is actually a useful path)

Reply:
Nice catch - int2fp should fold undef to zero - I'll just early out for now